### PR TITLE
Add persona stats

### PR DESCRIPTION
**Pull request by hagen-danswer**
## Description
[Provide a brief description of the changes in this PR]

image

## Backporting (check the box to trigger backport action)
Note: You have to check that the action passes, otherwise resolve the conflicts manually and tag the patches.
- [ ] This PR should be backported (make sure to check that the backport attempt succeeds)

### CODE REVIEW

File: backend/ee/danswer/server/analytics/api.py [hunk 1]
```diff
@@ -115,3 +118,78 @@ def get_danswerbot_analytics(
...
+def get_persona_messages(
+    persona_ids: str = Query(...),  # ... means this parameter is required
+    start: datetime.datetime | None = None,
+    end: datetime.datetime | None = None,
+    _: User | None = Depends(current_admin_user),
+    db_session: Session = Depends(get_session),
+) -> list[PersonaMessageAnalyticsResponse]:
+    """Fetch daily message counts for multiple personas within the given time range."""
+    # Convert comma-separated string to list of integers
+    parsed_persona_ids = [
```
Comment (pablonyx) [inline backend/ee/danswer/server/analytics/api.py]: Would prefer to directly type as list[str/int] (believe FastAPI supports enforcing this directly

File: backend/ee/danswer/db/analytics.py [hunk 2]
```diff
@@ -170,3 +170,67 @@ def fetch_danswerbot_analytics(
     )
 
     return results
+
+
+def fetch_persona_message_analytics(
+    db_session: Session,
+    persona_id: int,
```
Comment (pablonyx) [inline backend/ee/danswer/db/analytics.py:177]: Probably worth combining shared logic across the functions here

File: backend/ee/danswer/db/analytics.py [hunk 3]
```diff
@@ -170,3 +170,67 @@ def fetch_danswerbot_analytics(
...
+            ),
+            ChatMessage.time_sent >= start,
+            ChatMessage.time_sent <= end,
+            ChatMessage.message_type == MessageType.ASSISTANT,
+        )
+        .group_by(cast(ChatMessage.time_sent, Date))
+        .order_by(cast(ChatMessage.time_sent, Date))
+    )
+
+    return list(db_session.execute(query).all())  # type: ignore
```
Comment (pablonyx) [inline backend/ee/danswer/db/analytics.py]: Is type ignore necessary here?

File: backend/ee/danswer/server/analytics/api.py [hunk 4]
```diff
@@ -115,3 +118,78 @@ def get_danswerbot_analytics(
...
+        int(id.strip()) for id in persona_ids.split(",") if id.strip()
+    ]
+    persona_message_counts = []
+    start = start or (datetime.datetime.utcnow() - datetime.timedelta(days=30))
```
Comment (pablonyx) [inline backend/ee/danswer/server/analytics/api.py]: For any magic numbers would prefer to declar a constant somewhere that we reference

File: web/src/app/ee/admin/performance/lib.ts [hunk 5]
```diff
@@ -97,3 +97,82 @@ export function getDatesList(startDate: Date): string[] {
...
+}
+
+export interface PersonaSnapshot {
+  id: number;
+  name: string;
+  description: string;
+  is_visible: boolean;
+  is_public: boolean;
+}
+
```
Comment (pablonyx) [inline web/src/app/ee/admin/performance/lib.ts:114]: Would prefer using the existing Assistants provider

**Review by pablodanswer — CHANGES_REQUESTED** [review]

File: backend/ee/danswer/db/analytics.py (hunk 2 above)
Comment (hagen-danswer) [reply backend/ee/danswer/db/analytics.py:177]: tried it. its ugly, confusing, and probs not worth

**Review by pablodanswer — APPROVED** [review]

**Pull request merged**